tests/test_nb_run.py

```diff
@@ -133,3 +134,15 @@ def test_when_empty_then_succeeds(self, testdir2: Never):
         run = NotebookRun(nb, 300)
         res: NotebookResult = run.execute()
         assert res.error is None
+
+    def test_by_cell(self, testdir2: Never):
```

Comment:
I find it hard to understand what this test is doing, and that will make it harder to maintain this feature.

Please could we (a) Re-frame these tests in `test_when_x_then_y`, and (b) walk through a few more granular use cases, e.g. 
- when there are no tests in a notebook
- when there is 1 passing and 1 failing